live/root/etc/systemd/system/agama-dud.service

```diff
@@ -0,0 +1,31 @@
+[Unit]
+Description=Agama DUD
+
+After=network-online.target
+
+# and after we process agama params like info which can contain password
+After=agama-cmdline-process.service
+# the DUD should replace wrong self_update
+After=agama-self-update.service
+
+# before starting the Agama servers so they use the new packages
+Before=agama-web-server.service
+Before=agama.service
+Before=x11-autologin.service
+# before interactive password services
+Before=live-password-dialog.service
+Before=live-password-systemd.service
+
+[Service]
+Type=oneshot
+Environment=TERM=linux
+ExecStartPre=dmesg --console-off
+ExecStart=agama-dud
+ExecStartPost=dmesg --console-on
+TTYReset=yes
+TTYVHangup=yes
+StandardInput=tty
+TimeoutStartSec=infinity
```

Comment:
this seems to be a bit risky ... if this service blocks in some network call then starting whole installer can be blocked too due to that After / Before dependencies above

Reply:
This is copied from the self-update service written by @lslezak, so it's consistent.

It might be understood as risky, but we really need this to finish whatever it takes otherwise you'd have the system in an inconsistent way. And if some URL does not work / takes too long, then you don't want to continue anyway (and you should remove it from the command-line).

Reply:
A bit more info: DUD is a functionality for updating/extending the Live Media. It's also used for updating the Agama itself and of course other services that need to be updated. Blocking them till the update is done is a reasonable approach - better than restarting them later.

Reply:
if we resign on best effort approach in running installer then fine with me.

Reply:
This is my best effort :)